vm/default_runtime/src/lib.rs

```diff
@@ -0,0 +1,442 @@
+        // TODO: charge gas
+
+        let ret = {
+            // TODO: make its own method/struct
+            match to_actor.code {
+                x if x == *SYSTEM_ACTOR_CODE_ID => {
+                    actor::system::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                x if x == *INIT_ACTOR_CODE_ID => {
+                    actor::init::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                x if x == *CRON_ACTOR_CODE_ID => {
+                    actor::cron::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                x if x == *ACCOUNT_ACTOR_CODE_ID => {
+                    actor::account::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                x if x == *POWER_ACTOR_CODE_ID => {
+                    actor::power::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                x if x == *MINER_ACTOR_CODE_ID => {
+                    // not implemented yet
+                    // actor::miner::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    todo!()
+                }
+                x if x == *MARKET_ACTOR_CODE_ID => {
+                    // not implemented yet
+                    // actor::market::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                    todo!()
+                }
+                x if x == *PAYCH_ACTOR_CODE_ID => {
+                    actor::paych::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                x if x == *MULTISIG_ACTOR_CODE_ID => {
+                    actor::cron::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                x if x == *REWARD_ACTOR_CODE_ID => {
+                    actor::cron::Actor.invoke_method(&mut *runtime, *method_num, msg.params())
+                }
+                _ => todo!("Handle unknown code cids"),
+            }
+        };
+        return ret;
+    }
+    Ok(Serialized::default())
+}
+
+/// Transfers funds from one Actor to another Actor
+fn transfer<ST>(state: &ST, from: &Address, to: &Address, value: &TokenAmount) -> Result<(), String>
+where
+    ST: StateTree,
+{
+    if from == to {
+        return Ok(());
+    }
+    if value < &0u8.into() {
+        return Err("Negative transfer value".to_owned());
+    }
+    let mut from_actor = state
+        .get_actor(&from)
+        .map_err(|e| format!("transfer failed: {}", e))?
+        .ok_or_else(|| "transfer failed could not retrieve from actor".to_owned())?;
+    let mut to_actor = state
+        .get_actor(&to)
+        .map_err(|e| format!("transfer failed: {}", e))?
+        .ok_or_else(|| "transfer failed could not retrieve from actor".to_owned())?;
+
+    deduct_funds(&mut from_actor, &value)?;
+    deposit_funds(&mut to_actor, &value);
+    Ok(())
+}
```

Comment:
Where are these changes persisted? You're changing the local copy, but this does nothing

Reply:
fixed